llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -2205,7 +2205,24 @@ void SelectionDAGLegalize::ExpandFPLibCall(SDNode* Node,
     Results.push_back(Tmp.first);
     Results.push_back(Tmp.second);
   } else {
-    SDValue Tmp = ExpandLibCall(LC, Node, false).first;
+    bool IsSignedArgument = false;
+    switch (LC) {
+    case RTLIB::LDEXP_F32:
+    case RTLIB::LDEXP_F64:
+    case RTLIB::LDEXP_F80:
+    case RTLIB::LDEXP_F128:
+    case RTLIB::LDEXP_PPCF128:
```

Comment:
It might be a bit cleaner to check this as `Node->getOpcode() == ISD::FLDEXP`, to cover all libcall variants at once.

Reply:
I like that, thanks!  Done.